packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/Camera.java

```diff
@@ -71,164 +77,250 @@
 }
 
-public class Camera {
+/**
+ * Note: at this time we do not implement zero shutter lag (ZSL) capture. This is a potential
+ * improvement we can use in the future. The idea is in a TEMPLATE_ZERO_SHUTTER_LAG capture session,
+ * the system maintains a ring buffer of images from the preview. It must be in full auto moved
+ * (flash, ae, focus, etc). When you capture an image, it simply picks one out of the ring buffer,
+ * thus capturing an image with zero shutter lag.
+ *
+ * <p>This is a potential improvement for the future. A good example is the AOSP camera here:
+ * https://android.googlesource.com/platform/packages/apps/Camera2/+/9c94ab3/src/com/android/camera/one/v2/OneCameraZslImpl.java
+ *
+ * <p>But one note- they mention sometimes ZSL captures can be very low quality so it might not be
+ * preferred on some devices. If we do add support for this in the future, we should allow it to be
+ * enabled from dart.
+ */
+class Camera implements CameraCaptureCallback.CameraCaptureStateListener {
   private static final String TAG = "Camera";
 
-  /** Timeout for the pre-capture sequence. */
-  private static final long PRECAPTURE_TIMEOUT_MS = 1000;
+  private static final HashMap<String, Integer> supportedImageFormats;
+
+  // Current supported outputs
+  static {
+    supportedImageFormats = new HashMap<>();
+    supportedImageFormats.put("yuv420", ImageFormat.YUV_420_888);
+    supportedImageFormats.put("jpeg", ImageFormat.JPEG);
+  }
+
+  /**
+   * Holds all of the camera features/settings and will be used to update the request builder when
+   * one changes.
+   */
+  private final CameraFeatures cameraFeatures;
 
   private final SurfaceTextureEntry flutterTexture;
-  private final CameraManager cameraManager;
-  private final DeviceOrientationManager deviceOrientationListener;
-  private final boolean isFrontFacing;
-  private final int sensorOrientation;
-  private final String cameraName;
-  private final Size captureSize;
-  private final Size previewSize;
   private final boolean enableAudio;
   private final Context applicationContext;
-  private final CamcorderProfile recordingProfile;
   private final DartMessenger dartMessenger;
-  private final CameraZoom cameraZoom;
-  private final CameraCharacteristics cameraCharacteristics;
+  private final CameraProperties cameraProperties;
+  private final CameraFeatureFactory cameraFeatureFactory;
+  private final Activity activity;
+  /** A {@link CameraCaptureSession.CaptureCallback} that handles events related to JPEG capture. */
+  private final CameraCaptureCallback cameraCaptureCallback;
+  /** A {@link Handler} for running tasks in the background. */
+  private Handler backgroundHandler;
+  /**
+   * This a callback object for the {@link ImageReader}. "onImageAvailable" will be called when a
+   * still image is ready to be saved.
+   */
+  private final ImageReader.OnImageAvailableListener onImageAvailableListener =
+      new ImageReader.OnImageAvailableListener() {
```

Comment:
Could this be extracted to a method called by the lambda, rather than having a 15-line function inlined with the variable declarations?